src/robotFactory.js

```diff
@@ -1,10 +1,74 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
```

Comment:
you can set defaults like this
```suggestion
  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
```